### DISCUSSION TITLE
articles/2024/migrate-poetry-to-uv/

### DISCUSSION THREAD

**Discussion by giscus** [loopwerk/loopwerk.io#20] [Article discussions]
# articles/2024/migrate-poetry-to-uv/

So, like me you’ve decided to switch from Poetry to uv, and now you’re wondering how to actually migrate your pyproject.toml file? You’ve come to the right place!

https://www.loopwerk.io/articles/2024/migrate-poetry-to-uv/

**Comment by lluisCM** [comment]
Hello, Thanks for your post. My only concern with UV is that in the resultant pyproject.toml file you haven't specified the backend build system. So will uv be capable of build and install the current project? 
Checking the uv documentation I understand that a build system should be provided to be able to properly build. So yet without a third party tool uv can't fully build the project, so in this case we should always need to use it together with pdm, poetry or hatchling etc... 
Is that right or am I wrong?
I'm referring to:
[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"

Thank you.

**Reply by kevinrenskers** [reply]
When you create a brand new project with `uv init` you also don't get a build system config, so I guess uv just doesn't need it?

```
[project]
name = "temp"
version = "0.1.0"
description = "Add your description here"
readme = "README.md"
requires-python = ">=3.13"
dependencies = []
```

You can definitely run the resulting project, I've upgraded multiple Django projects using this method and they run just fine. I don't know about publishing installable packages though.

**Comment by joshuataylor** [comment]
Yeah this is confusing, as you get a bunch of different answers.

You might not need one if you're not distributing your project, e.g you're making a library.

If you are distributing as a library (i.e someone installs it via `pip install`, you can just stick in whatever you want, and it'll fall back to setuptools.

dependency-groups are really nice. Poetry is really great when you have just a single project and you're not distributing it, it is really good at that.

uv seems to be filling in the gaps where Poetry is falling behind, especially as all the good ideas from Poetry and other tools are now being standardised as PEPs.

**Comment by frostming** [comment]
Because I know many people are using other package managers, adding the import feature will make it easier for users to migrate to PDM with much less burden.

I'm glad this helps in your situation, even though it's not its intended use, haha.

This is the benefit of following standards, we use the same format(looking at Poetry)

**Reply by kevinrenskers** [reply]
PDM really saved the day, so thank you!

It's not ideal that Poetry doesn't follow the standards, it's definitely a downside. For me, who works on Python projects by himself, this wasn't a huge problem. But in teams it sucks that everyone is then forced to use Poetry.

Hopefully we get truly standardized project and lock files at some point and everyone can use their own preferred package manager 🤞

**Comment by vcidst** [comment]
> Every section that starts with tool.poetry can be removed

*looks at `[tool.poetry.dependencies]`* These instructions imply that I should remove the section that lists all dependencies...

**Reply by kevinrenskers** [reply]
Yup. Because running `uvx pdm import pyproject.toml` added the dependencies to the `[project]` section.

**Reply by vcidst** [reply]
oh right, I didn't see that. I'm still left with `[[tool.pdm.source]]` section in the pyproject.toml

I also have sections for a bunch of tools my repo used,
- [tool.ruff]
- [tool.bandit]
- [tool.towncrier]

Do you leave those sections as is?

**Reply by kevinrenskers** [reply]
I don't know what the `[tool.pdm.source]` section is, I didn't have that. But yeah those other tool sections can of course be left alone.

**Comment by shininglove** [comment]
simple and straightforward, thanks!

**Comment by ikudjoi** [comment]
Found this today
https://github.com/mkniewallner/migrate-to-uv

**Reply by tgagor** [reply]
I can confirm that this one is much better and easier way. 
It was harder to change my habits than project :D
